lib/segment/src/segment/facet.rs

```diff
@@ -58,16 +53,18 @@ impl Segment {
 
                 Either::Left(iter)
             } else {
-                // go over the values and filter the points
-                // aka. read from facet index
-                //
-                // This is more similar to a full-scan, but we won't be hashing so many times.
-                context = payload_index.struct_filtered_context(filter);
+                // Use cardinality estimations for each value to estimate the counts
+                let cardinality_ratio =
+                    filter_cardinality.exp as f32 / (self.available_point_count() + 1) as f32;
 
+                // Go over each value in the map index
                 let iter = facet_index
-                    .iter_filtered_counts_per_value(&context)
+                    .iter_counts_per_value()
```

Comment:
Approving since there is actually a test for this path :)

Reply:
That is a good point @agourlay, I think it is still possible for count to be 0 here. I will handle that case.

There is actually no test exercising this path right now, only manual tests because we need a large dataset. I'll try to make one which covers this

Reply:
exercised this path in https://github.com/qdrant/qdrant/pull/5013/commits/19575bf92d743b599c4c3bb378f7fb1c37a9da39